### ISSUE TITLE
Possibility for local extensions/vsix packages?

### ISSUE THREAD

**Issue by crankedguy** [shanalikhan/code-settings-sync#778]
I have quite a few private extensions running which I won't place on the marketplace.
As only the extension names are synced and then downloaded/installed through CLI these extensions are always missed.

Any possibility to provide vsix packages maybe in the user directory from where they are uploaded to git and then downloaded and installed? Did you ever think of such a thing?
It is always a hassle as I have to sync them manually.

Therefore it is nearly the same work to do as if I transfer the whole extensions and user directory.

**Comment by shanalikhan** [comment]
Nice idea. 

What i think can be done is in global settings you can setup the path of `vsix` package of certain extensions. If Settings Sync is unable to find extension on marketplace it should look on those path and install using packages offline.

**Comment by shanalikhan** [comment]
Currently low priority for me between.
I'm open for PR and assist anyone who is willing to work and release new version.

**Comment by crankedguy** [comment]
> What i think can be done is in global settings you can setup the path of `vsix` package of certain extensions. If Settings Sync is unable to find extension on marketplace it should look on those path and install using packages offline.

Yes this seems like a great one! I did not find a setting for setting a path in the global settings, only found --extensions-dir to set the root path in general from command line.
Sadly I have no spare time available right now. I think what I would do would be something in this direction : A certain directory has to be created in user directory for the vsix packages code-settings-sync should sync. They would be sent to GitHub then. When syncing downwards they would just be pulled and installed from command-line with --install-extension command line.

Please update when you have time and work on it, even if low priority, this would be really handy.

Thanks

**Comment by ADIX7** [comment]
I'm currently working on this. My idea is that vsix files can be stored in the gist as base64 files. I'm new to vscode extension development, @shanalikhan can you tell me how can I create a temp file? I will save the vsix in a temp file and tell vscode to install that.

**Comment by shanalikhan** [comment]
@ADIX7 

Is there any update ? I'm available to assist you over here or on slack. Feel free to ask questions.
